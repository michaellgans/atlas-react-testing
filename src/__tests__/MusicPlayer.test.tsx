// Tests the MusicPlayer Component

// Imports
import { render } from "@testing-library/react";
import MusicPlayer from "@/MusicPlayer";
import { expect, test } from "vitest";

test("MusicPlayer renders children correctly", () => {
    // Generate mock children
    const { getByText } = render(
        <MusicPlayer>
            <div>
                Mock Child
            </div>
        </MusicPlayer>
    );

    // Child should be rendered inside Music Player
    expect(getByText("Mock Child")).toBeDefined();
});

test("MusicPlayer renders correctly and matches snapshot", () => {
    const { asFragment } = render(
        <MusicPlayer>
            <div>
                Mock Child
            </div>
        </MusicPlayer>
    );

    // Take Snapshot
    expect(asFragment()).toMatchSnapshot();
})