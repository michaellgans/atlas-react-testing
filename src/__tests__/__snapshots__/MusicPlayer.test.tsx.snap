// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MusicPlayer renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="music-player dark:bg-dark-player-background bg-player-background flex flex-col md:flex-row max-w-4xl justify-center rounded-lg shadow-lg"
  >
    <div>
      Mock Child
    </div>
  </div>
</DocumentFragment>
`;
